### ISSUE TITLE
feat[train]: Set up PyTorch Lightning module

### ISSUE THREAD

**Issue by Hamza-Bin-Aamir** [Hamza-Bin-Aamir/PIVOT#28]
Implement LightningModule wrapper
Define training_step with multi-task loss
Define validation_step
Configure optimizers and schedulers
Add logging hooks
